app/validators/questionnaire_schema.py

```diff
@@ -503,6 +507,32 @@ def get_calculation_block_ids(*, block: Mapping, source_type: str) -> list[str]:
             if source[1]["source"] == source_type
         ]
 
+    def get_answer_ids_for_value_source(
+        self, value_source: Mapping[str, str]
+    ) -> list[str]:
+        """
+        Gets the list of answer_ids relating to the provided value source. Either the identifier if its an answer source
+        or the list of included answer ids in the case of a calculated or grand calculated summary
+        """
+        source = value_source.get("source")
```

Comment:
My 2 cents:
> I am not a fan of how defined_value becomes value_source when this method is invoked

The current behaviour is appropriate, `_get_numeric_value` has a variable that can both be an integer or a dict (value source), when we know it is  value source, it is passed in the context of a method that is expecting to work with value sources. The source of the issue is because `_get_numeric_value` has more than one responsibility. Could use function overloading to handle each case independently but seems overkill for validator given there is already precedence.

> since we only want to pass what we want to use in a given method, can we pass identifier and value source as arguments?

The function does only pass in what it needs?.

Let me know if I have misunderstood the comment.

Reply:
What I mean is, you only need `source` and `identifier` in this method below, you don't need the whole "value source" dict.  But I get `defined_value` can be both dict or integer, my mistake, so disregard that first comment.

Reply:
but if the value source dict only contains source and identifier why take them out?

Reply:
Also, if identifier and source always exist here we should access them using the [key] instead of `.get()` but it's all just details and non blockers.

Reply:
> but if the value source dict only contains source and identifier why take them out?

For clarity/readability but yeah, just keep it as it is.

Reply:
Happy to switch to `[key]` though, done